### ISSUE TITLE
Uniformize log rotation/archiving

### ISSUE THREAD

**Issue by landryb** [georchestra/georchestra#1268]
Right now, by default all logs but 3 are rotated daily:
- geonetwork.log : rotated only once, not archived
- geoserver.log : rotated only once, not archived
- cas-webapp-server.log : rotated, not archived/compressed

```
-rw-r--r-- 1 tomcat tomcat   103618 Feb 18 14:36 /srv/log/tomcat/cas-server-webapp.log
-rw-r--r-- 1 tomcat tomcat   182232 Feb  3 19:41 /srv/log/tomcat/cas-server-webapp.log.2016-02-03
-rw-r--r-- 1 tomcat tomcat   404653 Feb  4 19:17 /srv/log/tomcat/cas-server-webapp.log.2016-02-04
-rw-r--r-- 1 tomcat tomcat    13162 Feb  7 19:16 /srv/log/tomcat/cas-server-webapp.log.2016-02-07
-rw-r--r-- 1 tomcat tomcat    58632 Feb  8 17:44 /srv/log/tomcat/cas-server-webapp.log.2016-02-08
-rw-r--r-- 1 tomcat tomcat   107852 Feb  9 16:36 /srv/log/tomcat/cas-server-webapp.log.2016-02-09
-rw-r--r-- 1 tomcat tomcat    86920 Feb 10 18:21 /srv/log/tomcat/cas-server-webapp.log.2016-02-10
-rw-r--r-- 1 tomcat tomcat    47483 Feb 11 17:57 /srv/log/tomcat/cas-server-webapp.log.2016-02-11
-rw-r--r-- 1 tomcat tomcat    80123 Feb 12 17:16 /srv/log/tomcat/cas-server-webapp.log.2016-02-12
-rw-r--r-- 1 tomcat tomcat    13162 Feb 13 17:28 /srv/log/tomcat/cas-server-webapp.log.2016-02-13
-rw-r--r-- 1 tomcat tomcat   154957 Feb 16 20:11 /srv/log/tomcat/cas-server-webapp.log.2016-02-16
-rw-r--r-- 1 tomcat tomcat    47726 Feb 19 14:55 /srv/log/tomcat/geonetwork.log
-rw-r--r-- 1 tomcat tomcat 10488586 Feb 19 14:54 /srv/log/tomcat/geonetwork.log.1
-rw-r--r-- 1 tomcat tomcat  1915724 Feb 19 14:15 /srv/log/tomcat/geoserver.log
-rw-r--r-- 1 tomcat tomcat 10490609 Feb 18 10:37 /srv/log/tomcat/geoserver.log.1
```

The others are rotated/archived daily, so would be nice to have a coherent config for everything.

For archiving/daily rotation to work fine i also need to add apache-log4j-extras dependency via pom.xml files, and modify the default log4j config.

**Comment by landryb** [comment]
Oh and according to http://stackoverflow.com/questions/5117758/configuring-rollingfileappender-in-log4j - i should bump the log4j dependency in geoserver to 1.2.16+ - by default we have 1.2.14 :/

**Comment by landryb** [comment]
And according to http://stackoverflow.com/questions/5117758/configuring-rollingfileappender-in-log4j/13037467#13037467 i also need at least log4j 1.2.14 for geonetwork otherwise its dtd blows because rollingPolicy isnt defined.
```
log4j:WARN Continuable parsing error 7 and column 80
log4j:WARN Element type "rollingPolicy" must be declared.
log4j:WARN Continuable parsing error 13 and column 16
```
Using 1.2.17 works. Yay java.

**Comment by landryb** [comment]
I didnt see the last issue on my dev platform, probably because the latter has geonetwork in tomcat-georchestra and a newer log4j (or dtd) was already loaded, on my production instance geonetwork is in its own tomcat and it failed there...

**Comment by landryb** [comment]
https://github.com/geonetwork/core-geonetwork/pull/1451 for the latter, i still need to remerge 3.0.x in our fork.

**Comment by fvanderbiest** [comment]
I think we can close this one. OK ?

**Comment by landryb** [comment]
I .... *think* so. fuzzy memory.
